core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -1519,8 +1533,17 @@ wasm_instantiate(WASMModule *module, bool is_sub_inst, uint32 stack_size,
         (WASMModuleInstanceCommon *)module_inst);
 #endif
     (void)global_data_end;
+
     return module_inst;
+
 fail:
+#if WASM_ENABLE_DEBUG_INTERP != 0
+    if (!is_sub_inst) {
+        os_mutex_lock(&module->ref_count_lock);
+        module->ref_count--;
```

Comment:
Seems no need to decrease it here, the following wasm_instantiate decreate it again

Reply:
1: Yes, you're right. I move these line to the place where malloc memory for Instance failed.
2: AoTModule is not related to interpreter debugging, I'll consider that when refactoring AOT debugging, that is in lower priority currently.